src/types.jl

```diff
@@ -67,13 +72,15 @@ including `*`, `+`, `-`, `/`, `^`, `sqrt`, and `cbrt`.
 - `dimensions::Dimensions`: dimensions of the quantity
 - `valid::Bool`: whether the quantity is valid or not
 """
-struct Quantity{T}
+struct Quantity{T, R}
     value::T
-    dimensions::Dimensions
+    dimensions::Dimensions{R}
     valid::Bool
 
-    Quantity(x; kws...) = new{typeof(x)}(x, Dimensions(; kws...), true)
-    Quantity(x, valid::Bool; kws...) = new{typeof(x)}(x, Dimensions(; kws...), valid)
-    Quantity(x, d::Dimensions) = new{typeof(x)}(x, d, true)
-    Quantity(x, d::Dimensions, valid::Bool) = new{typeof(x)}(x, d, valid)
+    Quantity(x; kws...) = new{typeof(x), SimpleRatio{SafeInt}}(x, Dimensions(; kws...), true)
```

Comment:
Could `SimpleRatio{SafeInt}` be declared as a global `const`? Then it's easier to change the default.